### ISSUE TITLE
Make OidcTestSecurityIdentityAugmentor WAY faster by making privateKey's generation final and static (thus unique for the test execution)

### ISSUE THREAD

**Issue by YassinHajaj** [quarkusio/quarkus#38078]
### Description

The performance issue with tests utilizing `@TestSecurity` in conjunction with `@OIDCSecurity` was observed in the [provided reproducer project](https://github.com/YassinHajaj/quarkus-slow-test-security). The test run times are as follows (pattern is `HH:mm:ss.SSS`)

- `testWithoutSecurity` : Start at 01:15:28.516, End at 01:15:29.080
- `testWithSecurity` : Start at 01:15:29.080, End at 01:17:03.021
- `testWithCustomSecurity` : Start at 01:17:03.022, End at 01:17:03.180

**You'll notice that the `testWithSecurity` takes a little less than 2 minutes to run 1000 empty tests.**

The root cause of the slowdown is attributed to the repeated instantiation of `OidcTestSecurityIdentityAugmentor` for each test. This is due to the way the augment method in the `OidcTestSecurityIdentityAugmentor` class is written

```java
private SecurityIdentity augment(SecurityIdentity identity, Annotation[] annotations) {
    Instance<TestSecurityIdentityAugmentor> producer = CDI.current().select(TestSecurityIdentityAugmentor.class);
    if (producer.isResolvable()) {
        return producer.get().augment(identity, annotations);
    }
    return identity;
}
```

The bottleneck is specifically the generation of the `privateKey` field inside of the `OidcTestSecurityIdentityAugmentor`'s constructor

```java
privateKey = KeyUtils.generateKeyPair(2048).getPrivate();
```

This key generation method is computationally intensive and hence slow.

### Implementation ideas

To address this issue, I propose modifying the class to initialize the `privateKey` statically. This change ensures that the key is generated only once, reducing the overhead for each test:

```java
private static final PrivateKey privateKey;

static {
    try {
        privateKey = KeyUtils.generateKeyPair(2048).getPrivate();
    } catch (NoSuchAlgorithmException ex) {
        throw new RuntimeException(ex);
    }        
}
```

This modification should significantly improve the performance of the tests.

**Comment by YassinHajaj** [comment]
I have created a pull request already : https://github.com/quarkusio/quarkus/pull/38079

**Comment by sberyozkin** [comment]
Fixed by #38078
